engine/classes/ElggMenuItem.php

```diff
@@ -396,6 +399,39 @@ public function addLinkClass($class) {
 	}
 
 	/**
+	 * Set required AMD modules
+	 *
+	 * @param string[]|string $modules One or more required AMD modules
+	 * @return void
+	 */
+	public function setRequire($modules) {
+		$this->data['require'] = (array) $modules;
+	}
+
+	/**
+	 * Get required AMD modules
+	 *
+	 * @return string[]
+	 */
+	public function getRequire() {
+		$modules = (array) $this->data['require'];
+		return array_filter($modules, function($m) {
```

Comment:
Problematic because there are many way in which this can be registered: set
on menu item object property, through factory array or data etc
On Mar 19, 2016 9:44 PM, "Steve Clay" <notifications@github.com> wrote:

> In engine/classes/ElggMenuItem.php
> <https://github.com/Elgg/Elgg/pull/9520#discussion_r56754871>:
>
> > +	 *
> > +	 * @param string[]|string $modules One or more required AMD modules
> > +	 * @return void
> > +	 */
> > +	public function setRequire($modules) {
> > +		$this->data['require'] = (array) $modules;
> > +	}
> > +
> > +	/**
> > +	 * Get required AMD modules
> > +	 *
> > +	 * @return string[]
> > +	 */
> > +	public function getRequire() {
> > +		$modules = (array) $this->data['require'];
> > +		return array_filter($modules, function($m) {
>
> Validate while adding instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9520/files/31c9e05261f5a0949ddc3b9c6b77d622c092e3e3#r56754871>
>